packages/block-library/src/block/edit.js

```diff
@@ -8,26 +8,29 @@ import classnames from 'classnames';
  */
 import { useRegistry, useSelect, useDispatch } from '@wordpress/data';
 import { useRef, useMemo, useEffect } from '@wordpress/element';
-import { useEntityProp, useEntityRecord } from '@wordpress/core-data';
+import { useEntityRecord, store as coreStore } from '@wordpress/core-data';
 import {
 	Placeholder,
 	Spinner,
-	TextControl,
-	PanelBody,
+	ToolbarButton,
+	ToolbarGroup,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import {
 	useInnerBlocksProps,
 	__experimentalRecursionProvider as RecursionProvider,
 	__experimentalUseHasRecursion as useHasRecursion,
 	InnerBlocks,
-	InspectorControls,
 	useBlockProps,
 	Warning,
 	privateApis as blockEditorPrivateApis,
 	store as blockEditorStore,
+	BlockControls,
 } from '@wordpress/block-editor';
 import { getBlockSupport, parse } from '@wordpress/blocks';
```

Comment:
not sure if adding the `editor` package as a dependency of blocks is a good thing or not, but couldn't see another way of setting the rendering mode from within a block.

Reply:
One option that comes to mind, you could use a filter in the `edit-post` package to augment the block.

For example, you could pass in an additional prop to the block edit function, something like `editPatternInFocusMode`. The block could check for the existence of that prop, and when it exists show the button for editing and make the `onClick` trigger the function.

The functionality in the site editor could potentially be built the same way.

Reply:
ah, nice, will make that change tomorrow. I might be able to add the filter to the `editor` package as now that is also being shared by the site editor it might then work across both.

Reply:
Moving this to the editor package worked for both post and site editor